### ISSUE TITLE
Firewall restrictions through GUI don't affect ping

### ISSUE THREAD

**Issue by emanruse** [QubesOS/qubes-issues#8892]
### Qubes OS release

4.1.x
4.2

### Brief summary

Restricting the network connectivity of a qube through the Firewall tab in qube's GUI settings does not restrict `ping` - one can ping any host from the qube.

Restrict default ping to selected hosts if "Limit outgoing connections" is active. Make it user configurable through the GUI.

### Steps to reproduce

1. For some qube, open Firewall rules tab in the Settings GUI
2. Click the "+" button to add a new rule
3. Set

Address = example.com
Protocol = Any
Port/Service - leave it as is (i.e. Any)

4. Start a terminal in the qube and run:

```
ping example.com
ping fsf.org
```

### Expected behavior

Only ping to example.com should be possible. (because Protocol = any, Port/Service = any)

### Actual behavior

It is possible to ping any host.

**Comment by UndeadDevel** [comment]
Take a look at the bottom of that Firewall Settings page, where it says that ICMP (ping) and DNS are not restricted and to use the command line tool (`qvm-firewall`) for more granular control.

**Comment by marmarek** [comment]
> Should we also consider that it is actually more secure to apply the limitation, not to allow ICMP leaks?

https://www.qubes-os.org/doc/data-leaks/#the-role-of-the-firewall

**Comment by marmarek** [comment]
You missed the point. The firewall here is not to prevent leaks from a malicious VM. It's to prevent user mistakes like opening some random website in your banking qube.

**Comment by 3hhh** [comment]
I agree that the firewall GUI should make all options available on command-line to the user and not have that weird "DNS & ICMP allow" _hidden_ default.
To preserve backward compatibility it'd probably be best to keep it, but make it visible to and configurable by the user via the GUI.

**Comment by emanruse** [comment]
Affects 4.3

Please expose DNS and ICMP in settings for easier control. Have them in different color with a matching color in the "NOTE" but let's have them.
